libraries/AP_DAL/AP_DAL.cpp

```diff
@@ -87,6 +97,46 @@ void AP_DAL::start_frame(AP_DAL::FrameType frametype)
 #endif
 }
 
+/*
+  setup optional sensor backends
+ */
+void AP_DAL::init_sensors(void)
+{
+    init_done = true;
+    bool alloc_failed = false;
+
+    /*
+      we only allocate the DAL backends if we had at least one sensor
+      at the time we startup the EKF
+     */
+
+    auto *rangefinder = AP::rangefinder();
+    if (rangefinder && rangefinder->num_sensors() > 0) {
+        alloc_failed |= (_rangefinder = new AP_DAL_RangeFinder) == nullptr;
+    }
+
+    auto *airspeed = AP::airspeed();
+    if (airspeed != nullptr && airspeed->get_num_sensors() > 0) {
+        alloc_failed |= (_airspeed = new AP_DAL_Airspeed) == nullptr;
+    }
+
+    auto *beacon = AP::beacon();
+    if (beacon != nullptr && beacon->enabled()) {
+        alloc_failed |= (_beacon = new AP_DAL_Beacon) == nullptr;
+    }
+
+#if HAL_VISUALODOM_ENABLED
+    auto *visualodom = AP::visualodom();
+    if (visualodom != nullptr && visualodom->enabled()) {
+        alloc_failed |= (_visualodom = new AP_DAL_VisualOdom) == nullptr;
+    }
+#endif
+
+    if (alloc_failed) {
+        AP_BoardConfig::config_error("Unable to allocate DAL backends");
```

Comment:
In an extreme case where we enable EKF in flight - might this not lead to your vehicle crashing?

Reply:
you would have to not have any EKF enabled before takeoff. We need to think about whether we want to handle this case

Reply:
I can' think of a use case for doing that. At the moment the only platform that doesn't use EKF is AP_Periph and that doesn't build the EKF's anyway.